### DISCUSSION TITLE
How to detect entire scenes and not only shot changes?

### DISCUSSION THREAD

**Discussion by hellocaio** [Breakthrough/PySceneDetect#382] [General]
Hi guys,

I've been experimenting with different scene detection tools, mainly PySceneDetect and Google Video Intelligence. I noticed that what these tools call scene detection is really shot detection. They detect different shots, not scene. A scene, technically, is a sequence of shots, constituted of a setting (a place, time period, environment...), characters, events, dialogue, purpose and so on.

I would like to detect entire scenes. Is this possible? If so, how can this be done?

**Comment by yumianhuli1** [comment]
I think this library is from the Angle of shot view, not from the Angle of scene understanding trigger to cut
From the perspective of scene understanding, deep learning models should be used
